### PR TITLE
Sim Placement Service

### PR DESCRIPTION
**Pull request by kylestach**
## Description
Make ball and robot placement a service.

Allows programmatic setting of ball/robot locations, which is useful for quick-and-dirty testing scripts.

## Associated Issue
N/A

## Design Documents
N/A

### Placement
1. Move the ball in sim

Expected result: no regression

### CODE REVIEW

**Review by HussainGynai — COMMENTED** [review]
The code looks good, but I can't test since I think you branched off some old branch/commit that has some errors somewhere (in world_state_converter it says last change was in Feb, but in ros2 it says last change was 2 weeks ago).

**Review by kfu02 — APPROVED** [review]
Looks like nothing changed!

File: soccer/src/soccer/ui/sim_field_view.cpp [hunk 1]
```diff
@@ -107,45 +104,70 @@ void SimFieldView::mouseMoveEvent(QMouseEvent* me) {
...
-    if (_dragMode == DRAG_SHOOT) {
-        grSim_Packet simPacket;
-        grSim_BallReplacement* ball_replace = simPacket.mutable_replacement()->mutable_ball();
+void SimFieldView::drag_ball(const QPoint& screen_pos) {
+    set_ball_position(_screenToWorld * screen_pos);
+}
+
+void SimFieldView::set_ball_position(const rj_geometry::Point& field_pos) {
+    auto request = std::make_shared<rj_msgs::srv::SimPlacement::Request>();
+    request->ball.position.push_back(rj_convert::convert_to_ros(_screenToWorld * field_pos));
```
Comment (kylestach) [inline soccer/src/soccer/ui/sim_field_view.cpp]:
```suggestion
    request->ball.position.push_back(rj_convert::convert_to_ros(field_pos));
```

**Review by HussainGynai — APPROVED** [review]
LGTM

**Pull request merged**